### PR TITLE
56 heatmap rendering is invalid

### PR DESCRIPTION
**Pull request by DuchessMuhr**
## What

Moved the geoJson files into a local folder and updated these so that the location names correspond to what we get from the API. Updated the colors and legend to be dynamic with the highest number of reported crimes for a specific crime, divided into sixths.

## How
GeoJson: Deleted the fetching file and replaced them with local json files. The legend updates a variable with the highest reported number when creating the dictonary. This variable is then used for calculating the limits when creating the legend and assigning color.

## Why
To make the heatmap more understandable and colors actually meaning something when looking at the map.

### Notes
At the moment the maxCrime variable is based both on regions and municipalities which in turn results in a very bleak looking map when looking at regions and fire since the max value for fire in municipalities is so high.

### CODE REVIEW

File: agile-app/src/app/map/page.tsx [hunk 1]
```diff
@@ -11,6 +10,9 @@ import {getCrimeData} from "@/scripts/dataFetching";
 import {CrimeData} from "@/scripts/dataFetching";
 import { GeoJSON as LeafletGeoJSON } from "leaflet";
 import MapLegend from "@/components/mapLegend/mapLegend";
+import regionData from "../../../public/geojson/geoJsonRegion.json";
+import municipalityData from "../../../public/geojson/geoJsonMunicipality.json";
+import { get } from "http";
```
Comment (willayy) [inline agile-app/src/app/map/page.tsx:15]: Remove unused import
```js
import { get } from "http";
```

File: agile-app/src/app/map/page.tsx [hunk 2]
```diff
@@ -1,7 +1,6 @@
 "use client"
 
 import {useEffect, useRef, useState} from "react";
-import {fetchMunicipalityData, fetchRegionData} from "@/scripts/geoFetching";
 import {GeoJSON, GeoJSONProps, MapContainer, TileLayer} from 'react-leaflet'
 import 'leaflet/dist/leaflet.css'
 import {Feature, GeoJsonObject, GeoJsonProperties, Geometry} from "geojson";
```
Comment (willayy) [inline agile-app/src/app/map/page.tsx:6]: remove unused import GeoJsonProperties
```suggestion
import {Feature, GeoJsonObject, GeoJsonProperties, Geometry} from "geojson";
``` -> ```suggestion
import {Feature, GeoJsonObject, Geometry} from "geojson";
```

File: agile-app/src/app/map/page.tsx [hunk 3]
```diff
@@ -1,7 +1,6 @@
 "use client"
 
 import {useEffect, useRef, useState} from "react";
-import {fetchMunicipalityData, fetchRegionData} from "@/scripts/geoFetching";
 import {GeoJSON, GeoJSONProps, MapContainer, TileLayer} from 'react-leaflet'
```
Comment (willayy) [inline agile-app/src/app/map/page.tsx:4]: Remove unused import

```suggestion
import {GeoJSON, GeoJSONProps, MapContainer, TileLayer} from 'react-leaflet'
``` -> ```suggestion
import {GeoJSON, MapContainer, TileLayer} from 'react-leaflet'
```

File: agile-app/src/app/map/page.tsx [hunk 4]
```diff
@@ -26,6 +28,20 @@ interface NumberDictionary {
 type CustomFeature = Feature<Geometry, CustomFeatureProperties> | undefined
 type Crimes = CrimeData[]
 
+const defaultLegendItems = [
```
Comment (willayy) [inline agile-app/src/app/map/page.tsx:31]: This variable is declared but never read, suggesting you remove it

File: agile-app/src/app/map/page.tsx [hunk 5]
```diff
@@ -55,37 +71,53 @@ export default function Map() {
...
      * @param density - The density of crime events.
      * @returns The color code for rendering the feature.
      */
 
+    // function setMaxCrimeValue() {
+    //     maxCrimeValue = Math.max(...Object.values(getEventsOnType(selectedOptionCrime)));
+      
+    // }
+
     function getColor(density : number) {
```
Comment (willayy) [inline agile-app/src/app/map/page.tsx:94]: Make this more extendable by using a loop that iterates over intervals or a map to define the colors. You could even move out the color intervals and declare it as a constant outside the function.
```suggestion
    function getColor(density : number) {
        let maxCrimeSixth = maxCrimeValue/6;
        return (density > maxCrimeValue-1 ? '#b30000' :
            density > maxCrimeSixth*5 ? '#e34a33' :
                density > maxCrimeSixth*4 ? '#fc8d59' :
                    density > maxCrimeSixth*3 ? '#fdbb84' :
                        density > maxCrimeSixth*2 ? '#fdd49e' :
                            density > maxCrimeSixth ? '#fef0d9' :
                                '#FFFFFF')
    }
``` -> ```suggestion
    function getColor(density : number) {
        const colorIntervals: [number, string][] = [
        [6, '#b30000'],
        [5, '#e34a33'],
        [4, '#fc8d59'],
        [3, '#fdbb84'],
        [2, '#fdd49e'],
        [1, '#fef0d9'],
        [0, '#FFFFFF']
        ];
        for (const [threshold, color] of colorMap) {
            if (density > maxCrimeSixth * threshold) {
                return color;
            }
        }
    }
```

File: agile-app/src/app/map/page.tsx [hunk 6]
```diff
@@ -55,37 +71,53 @@ export default function Map() {
...
      * Determines the color style for rendering GeoJSON features based on the density of crime events.
      * 
      * @param density - The density of crime events.
      * @returns The color code for rendering the feature.
      */
 
+    // function setMaxCrimeValue() {
+    //     maxCrimeValue = Math.max(...Object.values(getEventsOnType(selectedOptionCrime)));
+      
+    // }
```
Comment (willayy) [inline agile-app/src/app/map/page.tsx:92]: Remove out commented code

File: agile-app/src/app/map/page.tsx [hunk 7]
```diff
@@ -55,37 +71,53 @@ export default function Map() {
...
+        return (density > maxCrimeValue-1 ? '#b30000' :
+            density > maxCrimeSixth*5 ? '#e34a33' :
+                density > maxCrimeSixth*4 ? '#fc8d59' :
+                    density > maxCrimeSixth*3 ? '#fdbb84' :
+                        density > maxCrimeSixth*2 ? '#fdd49e' :
+                            density > maxCrimeSixth ? '#fef0d9' :
                                 '#FFFFFF')
     }
     
+    function getLegendItems() {
```
Comment (willayy) [inline agile-app/src/app/map/page.tsx:105]: You could also make this an iteration that loops through the colorIntervals elements. This way you can add as many colorIntervals as you want without adding extra lines of code.

File: agile-app/src/scripts/geoFetching.ts
Comment (willayy) [inline agile-app/src/scripts/geoFetching.ts:1]: Why was this removed, im guessing we are storing the data locally?

Comment (DuchessMuhr) [reply agile-app/src/scripts/geoFetching.ts:1]: Thats right

**Comment by DuchessMuhr** [comment]
@willayy i have now removed the unused imports, variables and added missing commenting.

File: agile-app/src/app/map/page.tsx (hunk 5 above)
Comment (jmarkusson) [reply agile-app/src/app/map/page.tsx:82]: This has now been fixed

File: agile-app/src/app/map/page.tsx (hunk 7 above)
Comment (jmarkusson) [reply agile-app/src/app/map/page.tsx:106]: yes

**Review by Aoc67310 — APPROVED** [review]
